File: 2024/1/main.py
import sys

data = open(sys.argv[1]).read().strip()
lines = data.split('\n')

lefts, rights = [], []
for line in lines:
    left, right = line.split('   ')
    lefts.append(int(left))
    rights.append(int(right))

sorted_lefts = sorted(lefts)
sorted_rights = sorted(rights)

print(sum(abs(sl - sr) for sl, sr in zip(sorted_lefts, sorted_rights)))
print(sum(sl * sorted_rights.count(sl) for sl in sorted_lefts))

File: 2024/1/main.clj
(require '[clojure.string :as str])

(def data (slurp (first *command-line-args*)))

(def lines (str/split data #"\n"))

(def pairs (->> lines
                (map #(str/split % #"   "))))

(def lefts (sort (map #(Integer/parseInt (first %)) pairs)))
(def rights (sort (map #(Integer/parseInt (second %)) pairs)))

(println (reduce + (map #(abs (- %1 %2)) lefts rights)))
(println (reduce + (map #(* % ((frequencies rights) % 0)) lefts)))